File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left = 0
        right = len(heights) - 1
        max_area = 0
        while left < right:
            if max_area < ((right - left) * min(heights[left], heights[right])):
                max_area = (right - left) * min(heights[left], heights[right])
                if heights[left] <= heights[right]:
                    left += 1
                    continue
                else:
                    right -= 1
                    continue
            if heights[left] <= heights[right]:
                left += 1
                continue
            else:
                right -= 1
                continue
        return max_area

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-0.java
class Solution {
    public int evalRPN(String[] tokens) {
        Deque<Integer> stack = new ArrayDeque<>();
        
        for (String x : tokens){
            if (x.equals("+")){
                int right = stack.pop();
                int left = stack.pop();
                int sum = left + right;
                stack.push(sum);
            }else if (x.equals("-")){
                int right = stack.pop();
                int left = stack.pop();
                int sum = left - right;
                stack.push(sum);
            } else if (x.equals("*")){
                int right = stack.pop();
                int left = stack.pop();
                int sum = left * right;
                stack.push(sum);
            }else if (x.equals("/")){
                int right = stack.pop();
                int left = stack.pop();
                int sum = left / right;
                stack.push(sum);
            } else stack.push(Integer.parseInt(x));
        }
        return stack.pop();
    }
}
